packages/effect/src/Micro.ts

```diff
@@ -144,6 +144,45 @@ export interface MicroIterator<T extends Micro<any, any, any>> {
   next(...args: ReadonlyArray<any>): IteratorResult<YieldWrap<T>, Micro.Success<T>>
 }
 
+/**
+ * @since 3.9.0
+ * @category models
+ */
+export interface MicroPrimitive {
+  new<A, E = never, R = never>(): Micro<A, E, R>
+}
+
+// ----------------------------------------------------------------------------
+// Microable
+// ----------------------------------------------------------------------------
+
+const MicroProto = {
+  ...Effectable.EffectPrototype,
+  _op: "Micro",
+  [TypeId]: {
+    _A: identity,
+    _E: identity,
+    _R: identity
+  },
+  [Symbol.iterator]() {
+    return new SingleShotGen(new YieldWrap(this)) as any
+  }
+}
+
+const MicroBase: MicroPrimitive = (function() {
+  function Base() {}
+  Base.prototype = MicroProto
+  return Base as any
+})()
+
+/**
+ * @since 3.9.0
+ * @category constructors
+ */
+export abstract class MicroableClass<out A, out E = never, out R = never> extends MicroBase<A, E, R> {
```

Comment:
I think `Class` is fine. `MicroableClass` is a bit of a mouthful,  and the usage could be:

```ts
import { Micro } from "effect"

class Thing extends Micro.Class<A, E> {
  // ...
}
```